### ISSUE TITLE
Update content from the old cookbook

### ISSUE THREAD

**Issue by jenweber** [ember-learn/cookbook#6]
In very old versions of the guides, there are cookbook articles. Which of these can we update and reused? Comment with the topic you are researching/updating.

You can see the list here: https://guides.emberjs.com/v1.12.0/cookbook/

Articles should follow the template for our cookbook: https://github.com/ember-learn/cookbook/blob/main/guides/forms/template.md

**Comment by JennyJudova** [comment]
Looking at the old cook books all of the 'Contributing' section articles can be reused.

The rest I am not sure are even issues that need cook books? I think this is where going through the forum discussions and Stack overflow might be a good idea. 

I am happy to start moving the 'Contributing' section from the old cook book to this cook book shall we add it in this issue or create a separate one?

**Comment by locks** [comment]
> The rest I am not sure are even issues that need cook books? I think this is where going through the forum discussions and Stack overflow might be a good idea.

I tend to agree. I did a quick review of the recipes with my opinion on whether they're useful to keep.
Legend:
- 👍 Keep
- 👎 Don't keep
- ❓ Not sure whether to keep or not. Could be useful.

## User Interface & Interaction

👎 "Adding CSS Classes to Your Components" with Glimmer components the instruction here is to use the template.
👎 "Adding CSS Classes to Your Components Based on Properties" same as the above. Also covered in the [Conditional Content](https://guides.emberjs.com/v4.1.0/components/conditional-content/) section of the Guides.
❓ "Focusing a Textfield after It's Been Inserted" could be a recipe as there's some confusion around render lifecycle modifiers and modifiers in general. Would need an accessibility review to make sure we're not suggesting and anti-pattern.
👎 "Displaying Formatted Dates With Moment.js" moment is no longer the date manipulation library du jour, and this should be handled by an addon/library documentation.
👎 "Specifying Data-Driven Areas of Templates That Do Not Need To Update" On top of the engine optimizations that lessen this problem, you can make use of untracked/static data for this. `unbound` is not something we want to recommend.
❓ "Using Modal Dialogs" there are [community addons](https://emberobserver.com/?query=modal) to address this. It is a thorny issue and this recipe could be updated to make use of `in-element` and `RouteService#routeDidChange`, but I'm not sure it is worth it other recommending one of the addons.
❓ "Resetting scroll on route changes" there are once again some [addon solutions for this](https://emberobserver.com/?query=scroll). I'm not sure we want to advocate for this solution? There's also accessibility concerns around routing and this sort of functionality.

## Event Handling & Data Binding

👎 "Binding Properties of an Object to Its Own Properties" We cover getter properties in the Guides already.

## Helpers & Components

👎 "Creating Reusable Social Share Buttons" I think we share the necessary knowledge in the Components section of the guide. Furthermore, I think the solution proposed might have accessibility issues.
👎 "A Spinning Button for Asynchronous Actions" I think this sort of asynchrony should be solved with `ember-concurrency`. At most we would point to its documentation.
👎 "Adding Google Analytics Tracking" we have addons and many ways to do this. Furthermore, I don't think we're eager to feed the GA machine :P

## Working with Objects

👎 "Incrementing Or Decrementing A Property" doesn't really apply anymore, as we moved away from `EmberObject` APIs.
👎 "Setting Multiple Properties At Once" also does not apply for the same reason.
❓ "https://guides.emberjs.com/v1.12.0/cookbook/working_with_objects/continuous_redrawing_of_views/" is something that could still trip people up. That said, it might be as simple as pointing them to `ember-concurrency` and its documentation?

Doing this review raised a question for me: do we want to have recipes where we mostly recommend community addons and maybe have a small example and then link to the addon's documentation?

**Comment by JennyJudova** [comment]
@locks looks great, I agree with the overview and will look into the 'could be useful' articles
